components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -48,6 +48,10 @@ open class GleanInternalAPI internal constructor () {
 
     // `internal` so this can be modified for testing
     internal var initialized = false
+    // This is used to control metrics recording while still allowing sending of pings during glean
+    // initialization
+    internal var initializing = false
```

Comment:
Should we collapse this into a single enumerated value?  There's potential for introducing a new bug at some point in the future where we're in an undefined state (both `initializing` and `initialized` are `true`).

Reply:
`initializing` should only ever be used internal to the `initialize()` function (and tests) and there were only two states...  but I see the point behind this and made the change.  We may end up with more states in the future.